go-runtime/compile/schema.go

```diff
@@ -398,6 +402,60 @@ func visitCallExpr(pctx *parseContext, node *ast.CallExpr, stack []ast.Node) {
 	}
 }
 
+// validateCallExpr validates all function calls
+// checks if the is directly:
+// - calling external verbs
+// - publishing to an external module's topic
+func validateCallExpr(pctx *parseContext, node *ast.CallExpr, stack []ast.Node) {
+	selExpr, ok := node.Fun.(*ast.SelectorExpr)
+	if !ok {
+		return
+	}
+
+	var lhsIdent *ast.Ident
+	if expr, ok := selExpr.X.(*ast.SelectorExpr); ok {
+		lhsIdent = expr.Sel
+	} else if ident, ok := selExpr.X.(*ast.Ident); ok {
+		lhsIdent = ident
+	} else {
+		return
+	}
+
+	lhsObject := pctx.pkg.TypesInfo.ObjectOf(lhsIdent)
+	if lhsObject == nil {
+		return
+	}
+
+	var lhsPkgPath string
+	if pkgName, ok := lhsObject.(*types.PkgName); ok {
+		lhsPkgPath = pkgName.Imported().Path()
+	} else {
+		lhsPkgPath = lhsObject.Pkg().Path()
+	}
+
+	var lhsIsExternal bool
+	if !pctx.isPathInPkg(lhsPkgPath) {
+		lhsIsExternal = true
+	}
+
+	if lhsType, ok := pctx.pkg.TypesInfo.TypeOf(selExpr.X).(*types.Named); ok && lhsType.Obj().Pkg().Path() == ftlPkgPath {
+		// Calling a function on an FTL type
+		if lhsType.Obj().Name() == ftlTopicHandleTypeName && selExpr.Sel.Name == "Publish" {
```

Comment:
separately checking for `github.com/TBD54566975/ftl/go-runtime/ftl` and `TopicHandle` separately because the actual full name also includes the event type, which we don't care about